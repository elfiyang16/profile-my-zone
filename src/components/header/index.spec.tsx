import React from "react"
import { shallow } from "enzyme"
import Header, { IHeader } from "./"

describe("<Header />", () => {
  const defaultProps: IHeader = {
    header: "home",
  }

  let wrapper: any
  beforeEach(() => {
    wrapper = shallow(<Header {...defaultProps} />)
  })

  it("should render", () => {
    expect(wrapper).toMatchSnapshot()
  })

  it("should render burger menu ", () => {
    expect(wrapper.find(".site-header__burger-menu").length).toEqual(1)
  })
})
