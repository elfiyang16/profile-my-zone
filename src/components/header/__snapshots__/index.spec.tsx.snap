// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render 1`] = `
<header
  className="site-header "
>
  <div
    className="site-header__container"
  >
    <div
      className="site-header__main"
    >
      <div>
        <mockConstructor
          to="/"
        >
          <img
            alt="home logo"
            className="site-header__logo"
            src={Object {}}
          />
        </mockConstructor>
      </div>
      <nav
        className="site-header__responsive-menu-bar"
        onClick={[Function]}
        tabIndex={0}
      >
        <button>
          <img
            alt="menu logo"
            className="site-header__burger-menu"
            src={Object {}}
          />
        </button>
        <div
          className="site-header__menu"
        >
          <button
            className="site-header__menu-close"
          >
            <img
              alt="close"
              src={Object {}}
            />
          </button>
          <CustomLink
            item="Home"
            menu={false}
            setMenu={[Function]}
            url="/"
          />
          <CustomLink
            item="Blogs"
            menu={false}
            setMenu={[Function]}
            url="/#blogs"
          />
          <CustomLink
            item="About me"
            menu={false}
            setMenu={[Function]}
            url="/#about"
          />
          <CustomLink
            item="Contact"
            menu={false}
            setMenu={[Function]}
            url="/#contact"
          />
        </div>
      </nav>
    </div>
  </div>
</header>
`;
